### PR TITLE
Convert core project to TS and Heft

### PR DESCRIPTION
**Pull request by craigpalermo**
Follow-up work:
- Set up Storybook in Core project
- Restore test coverage to 100%
- Abstract Core's new eslint config into a preset so all projects can use it

### CODE REVIEW

File: apps/docs/package.json [hunk 1]
```diff
@@ -13,11 +13,11 @@
     "@mdx-js/tag": "^0.20.3",
     "is-absolute-url": "^2.1.0",
     "mdx-docs": "^1.0.0-9",
-    "pcln-design-system": "workspace:4.13.0",
+    "pcln-design-system": "workspace:4.13.1-beta.1-ts-core",
     "pcln-icons": "workspace:4.6.0",
-    "pcln-modal": "workspace:4.1.0",
```
Comment (sdalonzo) [inline apps/docs/package.json:18]: these should be *

File: tools/react-jesting-library/temp/_eslint-EXxU_Ulx.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (sdalonzo) [inline tools/react-jesting-library/temp/_eslint-EXxU_Ulx.json]: add temp to .gitignore

File: packages/core/src/utils/utils.spec.ts [hunk 3]
```diff
@@ -351,40 +350,14 @@ describe('utils', () => {
...
-        '{',
-        'smaaaallll css',
-        ';}',
-        '@media screen and (min-width:80em)',
-        '{',
-        'smaaaallll css',
-        ';}',
-      ]
-
+    test.only('returns the expected result when size is an array', () => {
```
Comment (sdalonzo) [inline packages/core/src/utils/utils.spec.ts]: no only

File: packages/core/src/Checkbox/Checkbox.stories.tsx [hunk 4]
```diff
@@ -128,25 +132,29 @@ Color.story = {
...
+    }
+    return (
       <>
         {/*
-            This example is for SC3
-              <Button dsRef={e => this.btnRef = e}>Click me</Button>
-          */}
+          This example is for SC3
+            <Button dsRef={e => this.btnRef = e}>Click me</Button>
+        */}
```
Comment (sdalonzo) [inline packages/core/src/Checkbox/Checkbox.stories.tsx:145]:
```suggestion
```

Comment (craigpalermo) [reply packages/core/src/Checkbox/Checkbox.stories.tsx:145]: Going to remove all forward ref demos later as discussed.

**Review by sdalonzo — APPROVED** [review]

**Review by pcln-barry — APPROVED** [review]

**Pull request merged**